mlos_bench/mlos_bench/storage/sql/trial.py

```diff
@@ -92,14 +77,19 @@ def _update(self, table: Table, timestamp: Optional[datetime],
                 conn.rollback()
                 raise
 
-    def update(self, status: Status,
-               metrics: Optional[Union[Dict[str, float], float]] = None
-               ) -> Optional[Dict[str, float]]:
-        metrics = super().update(status, metrics)
-        self._update(self._schema.trial_result, datetime.now(), status, metrics)
         return metrics
 
-    def update_telemetry(self, status: Status,
-                         metrics: Optional[Dict[str, float]] = None) -> None:
+    def update_telemetry(self, status: Status, metrics: List[Tuple[datetime, str, Any]]) -> None:
         super().update_telemetry(status, metrics)
-        self._update(self._schema.trial_telemetry, None, status, metrics)
+        for (timestamp, key, val) in metrics:
+            with self._engine.begin() as conn:
+                try:
+                    conn.execute(self._schema.trial_telemetry.insert().values(
+                        exp_id=self._experiment_id,
+                        trial_id=self._trial_id,
+                        ts=timestamp,
+                        metric_id=key,
+                        metric_value=None if val is None else str(val),
+                    ))
+                except IntegrityError as ex:
+                    _LOG.warning("Record already exists: %s :: %s", (timestamp, key, val), ex)
```

Comment:
Maybe also log the existing value if possible?

Reply:
Also makes me wonder if we need to consider the granularity of the timestamp to include `ms` or some such.

Reply:
as far as I know, `datetime` does include microseconds but it is up to the telemetry-generating scripts to decide on the granularity. In real applications, I believe we'll have telemetry data aggregated for 1 minute intervals or so. still, I want to use `datetime` in our API because this is Python standard and both SQLAlchemy and pandas handle it well